apps/vmq_enhanced_auth/src/vmq_enhanced_auth.erl

```diff
@@ -332,26 +382,31 @@ subst(MP, User, ClientId, [W | Rest], Acc) ->
 subst(_, _, _, [], Acc) ->
     lists:reverse(Acc).
 
-in(Type, User, Topic) when is_binary(Topic) ->
+merge_tuples(T1, T2) ->
+    list_to_tuple(tuple_to_list(T1) ++ tuple_to_list(T2)).
+
+in(Type, User, Topic, Label) when is_binary(Topic) ->
     TopicLen = byte_size(Topic) - 1,
     <<STopic:TopicLen/binary, _/binary>> = Topic,
     case validate(STopic) of
         {ok, Words} ->
             {Tbl, Obj} = t(Type, User, Words),
-            ets:insert(Tbl, Obj);
+            LabelObj = merge_tuples(Obj, {Label}),
+            ets:insert(Tbl, LabelObj);
         {error, Reason} ->
             error_logger:warning_msg("can't validate ~p acl topic ~p for user ~p due to ~p", [
                 Type, STopic, User, Reason
             ])
     end.
 
-insert_token(Type, _, Topic) when is_binary(Topic) ->
+insert_token(Type, _, Topic, Label) when is_binary(Topic) ->
     case parse_topic(Topic) of
         [] ->
             ok;
         Words ->
             {Tbl, Obj} = t(Type, token, Words),
-            ets:insert(Tbl, Obj)
+            LabelObj = merge_tuples(Obj, {Label}),
```

Comment:
Adding Label in t() function does not work because we have Label inside in() and insert_token().
but t() function is used inside all the check_*_acl() function where we don't have access to Label value so that's why we are skipping the t() function.

Reply:
t() used in check fns does not require topics & labels, so you can create two types of t()s, one which only returns table depending on type and token/user/pattern/all, the other one which returns both table & object to be stored.

Reply:
added the change to t() fns. to add label to Object without merge.